parser/src/grammar.lalrpop

```diff
@@ -0,0 +1,233 @@
+use std::iter;
```

Comment:
This file as broken - was trying to see if I could adapt the one from `parser2` to work with the current lexer